actors/miner/src/lib.rs

```diff
@@ -3682,38 +3706,110 @@ fn extend_sector_committment(
     curr_epoch: ChainEpoch,
     reward_stats: &ThisEpochRewardReturn,
     power_stats: &ext::power::CurrentTotalPowerReturn,
+    circulating_supply: &TokenAmount,
     new_expiration: ChainEpoch,
     sector: &SectorOnChainInfo,
     sector_size: SectorSize,
     claim_space_by_sector: &BTreeMap<SectorNumber, (u64, u64)>,
 ) -> Result<SectorOnChainInfo, ActorError> {
     validate_extended_expiration(policy, curr_epoch, new_expiration, sector)?;
+    let mut new_sector = sector.clone();
+    new_sector.expiration = new_expiration;
+    new_sector.power_base_epoch = curr_epoch;
 
-    // all simple_qa_power sectors with VerifiedDealWeight > 0 MUST check all claims
-    if sector.simple_qa_power {
-        extend_simple_qap_sector(
-            policy,
-            new_expiration,
-            curr_epoch,
-            reward_stats,
-            power_stats,
-            sector,
-            sector_size,
-            claim_space_by_sector,
-        )
-    } else {
-        extend_non_simple_qap_sector(new_expiration, curr_epoch, sector)
+    if !sector.simple_qa_power {
+        // Remove "spent" deal weights for non simple_qa_power sectors with deal weight > 0
+        new_sector.deal_weight = (&sector.deal_weight * (sector.expiration - curr_epoch))
+            .div_floor(&BigInt::from(sector.expiration - sector.power_base_epoch));
+
+        new_sector.verified_deal_weight = (&sector.verified_deal_weight
+            * (sector.expiration - curr_epoch))
+            .div_floor(&BigInt::from(sector.expiration - sector.power_base_epoch));
+    } else if sector.verified_deal_weight.is_positive() {
```

Comment:
I needed to share all the power updating code, so I inlined this code. But I'll probably put the code directly below below in a helper function.